resource_cache/resource_cache.go

```diff
@@ -221,7 +235,14 @@ func (o *ObjectCache) registerGVK(obj client.Object) {
 // blank object is stored in the cache it is imperative that the user of this function call Create
 // before modifying the obejct they wish to be placed in the cache.
 func (o *ObjectCache) Create(resourceIdent ResourceIdent, nn types.NamespacedName, object client.Object) error {
-	o.registerGVK(object)
+	if o.config.options.StrictGVK {
+		gvk, _ := utils.GetKindFromObj(o.scheme, object)
```

Comment:
Do we need to handle this error we're underscoring out or is it definitely never gonna pop?

Reply:
good point